qbg/vdp22cisco_oui.c

```diff
@@ -130,7 +130,7 @@ bool cisco_str2vdpnl_hndlr(struct vdpnl_oui_data_s *vdp_oui_p, char *token)
 				return false;
 			}
 			strncpy(v4_addr_str, token, data_len);
-			ret = inet_aton(v4_addr_str,
+			ret = inet_pton(AF_INET, v4_addr_str,
```

Comment:
`inet_pton()` does exhibit different behaviors than `inet_aton()` which should be examined. For one, where `inet_aton()` will return 1 for success and 0 for malformed address, `inet_pton()` has a third, possible return value of -1 on other error. The only documented error is if an invalid address family is specified and I don't see any other possible cases, but it still might be better to change the check from `if(!ret)` to something more like `if(ret <= 0)` just to future-proof the code.

The second behavior is that `inet_aton()` supports older address formats of A, A.B, and A.B.C, the first one being the case where the IPv4 address is provided as a single 32-bit number instead of the standard dotted-octet format. `inet_pton()` will return a zero for malformed address on these other cases. I don't think this should be an issue, but it should be considered in case there are other side-effects. `inet_aton()` is also not marked as deprecated like `inet_ntoa()` is, but it is nice to be consistent.

Reply:
> The second behavior is that `inet_aton()` supports older address formats of A, A.B, and A.B.C, the first one being the case where the IPv4 address is provided as a single 32-bit number instead of the standard dotted-octet format. `inet_pton()` will return a zero for malformed address on these other cases. I don't think this should be an issue, but it should be considered in case there are other side-effects. `inet_aton()` is also not marked as deprecated like `inet_ntoa()` is, but it is nice to be consistent.

Hmm, is the `token` an ipv4 dot string or a number? Because I saw `oui_vdp_hexstr2bin` just make the hex to bin and I can't find where the `.` added.

Reply:
I haven't taken the time yet to look at the code in more detail to see if it ever would come in as a number, but most likely it will always be in dotted-decimal form and there's no issue.

My overall concern with this PR is just making sure we don't break any existing behavior with changes that offer little to no gain. While `inet_ntoa()` is marked as deprecated, it's more to discourage future developers using it and making migrations to IPv6 more difficult in the future. However, it's not going to go away or break it's behavior anytime in the near or mid future. The function `inet_aton()` isn't even marked as deprecated, possibly because it has behaviors not covered in the newer functions. The use of `inet_ntop()` and `inet_pton()` over their predecessors only offer a real benefit when they are not hard-coded to `AF_INET`. Otherwise, they behave essentially the same (or close to it). I just want to make sure we aren't accepting in patches that don't ultimately improve things and actually end up breaking something in some less-often tested corner case.

Reply:
Fair enough. I'm OK to drop this PR. Thanks.